suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -122,81 +121,69 @@ export const DeviceList = ({ isVisible, onSelectDevice }: DeviceListProps) => {
         });
     };
 
-    //to hide/show the list with animation
     useEffect(() => {
-        if (isVisible) {
-            const otherDevicesHeight = notSelectedInstancelessDevices.length * ITEM_HEIGHT;
-
-            const separatorHeight =
-                hasUnselectedDevices && !isDiscoveryActive
-                    ? 2 * SEPARATOR_VERTICAL_PADDING + SEPARATOR_HEIGHT
-                    : 0;
-
-            const buttonHeight = isDiscoveryActive ? 0 : BUTTON_PADDING_TOP + BUTTON_HEIGHT;
-
-            const h =
-                MANAGER_MODAL_BOTTOM_RADIUS + //top margin for radius
-                PADDING_TOP +
-                otherDevicesHeight + // all other device
-                separatorHeight + // separator if there is one
-                buttonHeight +
-                PADDING_BOTTOM;
-
-            opacity.value = withDelay(100, withTiming(1, { duration: 200 }));
-            height.value = withTiming(h, { duration: 300 });
-
-            setIsShown(true);
-        } else {
-            opacity.value = withTiming(0, { duration: 300 });
-            height.value = withDelay(
-                100,
-                withTiming(0, { duration: 200 }, () => {
-                    runOnJS(setIsShown)(false);
-                }),
-            );
-        }
+        'worklet';
+
+        const otherDevicesHeight = notSelectedInstancelessDevices.length * ITEM_HEIGHT;
+
+        const separatorHeight =
+            hasUnselectedDevices && !isDiscoveryActive
+                ? 2 * SEPARATOR_VERTICAL_PADDING + SEPARATOR_HEIGHT
+                : 0;
+
+        const buttonHeight = isDiscoveryActive ? 0 : BUTTON_PADDING_TOP + BUTTON_HEIGHT;
+
+        const h =
+            MANAGER_MODAL_BOTTOM_RADIUS + //top margin for radius
+            PADDING_TOP +
+            otherDevicesHeight + // all other device
+            separatorHeight + // separator if there is one
+            buttonHeight +
+            PADDING_BOTTOM -
+            MANAGER_MODAL_BOTTOM_RADIUS;
+
+        height.value = isVisible ? h : 0;
```

Comment:
I know it was already like this before creating this PR. But When I looked at it now again, I realized it should be better to use `useDerivedValue`(https://www.reanimated2.com/docs/core/useDerivedValue) for both `height` and `opacity`. Then there would be no need for this effect and the animation would stay within the reanimated API borders.

IMO it will work just the same, but the readability will improve.

Reply:
based on the docs `useDerivedValue` should be used to derive sharedValue from other sharedValues. That is not what is being done here. Here I assign value based on other stuff

Reply:
Anyway. I have tried to use this and it works

Reply:
10e7175793

Reply:
Probably I did not make myself clear. The `useDerivedValue` is a hook that returns the shared value out of it. so iMO it would make sense create two derived values, each for `height` and `opacity`. 

```
const opacity = useDerivedValue(()=>{
  return  withTiming(isVisible ? 1 : 0, { duration: ANIMATION_DURATION });
});

const height = useDerivedValue(() => {
    const otherDevicesHeight = notSelectedInstancelessDevices.length * ITEM_HEIGHT;

        const separatorHeight =
            hasUnselectedDevices && !isDiscoveryActive
                ? 2 * SEPARATOR_VERTICAL_PADDING + SEPARATOR_HEIGHT
                : 0;
        const buttonHeight = isDiscoveryActive ? 0 : BUTTON_PADDING_TOP + BUTTON_HEIGHT;
        const h =
            MANAGER_MODAL_BOTTOM_RADIUS + //top margin for radius
            PADDING_TOP +
            otherDevicesHeight + // all other device
            separatorHeight + // separator if there is one
            buttonHeight +
            PADDING_BOTTOM -
            MANAGER_MODAL_BOTTOM_RADIUS;

     return withTiming(isVisible ? h : 0, { duration: ANIMATION_DURATION });

});
```

Like this, you don't need to set both values in a single useEffect. Hmu on Slack if you have any further questions or concers.

Reply:
As we agreed on the call, I will move the calculation into useAnimatedStyle, which will make more readable and the end game is to get rid of it in #12541

Reply:
https://github.com/trezor/trezor-suite/commit/bd4de936f5cb1388b9aaa6049db16f4d6c78cc1c